mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -1751,11 +1761,31 @@ void CMomentumPlayer::TimerCommand_Reset()
             // MOM_TODO do a trace downwards from the top of the trigger's center to touchable land, teleport the player there
             Teleport(&pCurrentZone->WorldSpaceCenter(), nullptr, &vec3_origin);
         }
-        else
+        return;
+    }
+
+    // Every other index is probably a stage (What about < 1 indexes? Mappers are weird and do "weirder"
+    // stuff so...)
+    CTriggerStage *pStage = nullptr;
+
+    while ((pStage = static_cast<CTriggerStage *>(gEntList.FindEntityByClassname(pStage, "trigger_momentum_timer_stage"))) != nullptr)
+    {
+        if (pStage && pStage->GetZoneNumber() == stage && pStage->GetTrackNumber() == track)
         {
-            Warning("Cannot reset, you have no current zone!\n");
+            // Teleport if we have a destination
+            if (const Vector *pVec = &pStage->GetAbsOrigin())
```

Comment:
There shouldn't ever be a stage zone without an origin, right? Were you running into this case somehow?

Reply:
Was a product of the old code so kept it in, though yeah should remove. I'll test without a check.

Reply:
Oh yeah jeez I see what you mean. It was a product of having it do 3 distinct things all collapsing into the same call, and by default, pVec was nullptr, so it was just verifying one of the 3 things it was going to (start, start mark, or stage trigger) actually got set into the variable. It can be removed here and simplified.